packages/discord.js-utilities/src/lib/PaginatedMessage.ts

```diff
@@ -170,11 +281,39 @@ export class PaginatedMessage {
 	];
 }
 
+/**
+ * @example
+ * ```typescript
+ * // To utilize actions you can use the [[IPaginatedMessageAction]] by implementing it into a class.
+ *
+ * class ForwardAction implements IPaginatedMessageAction {
+ *   public id = '▶️';
+ *
+ *   public run({ handler }) {
+ *     if (handler.index !== handler.pages.length - 1) ++handler.index;
+ *   }
+ * }
+ *
+ * // You can also give the object directly.
+ *
+ * const StopAction: IPaginatedMessageAction {
+ *   id: '⏹️',
+ *   disableResponseEdit: true,
+ *   run: ({ response, collector }) => {
+ *     await response!.reactions.removeAll();
+ *     collector!.stop();
+ *   }
+ * }```
+ */
 export interface IPaginatedMessageAction {
 	id: string;
+	disabledResponseEdit?: boolean;
 	run(context: PaginatedMessageActionContext): Awaited<unknown>;
```

Comment:
To be honest, I'm not a fan of doing this statically, there are two solutions:

- **Semver Minor**: We track the index state, if it changed, after the call, we edit the message. This option is more intuitive and easier to use.
- **Semver Major**: Make the function accept a value that tells whether or not the message should be refreshed.

Reply:
Ok, I get that I can change that right now.
I could probably implement both tbh.
Would the action's run method return a boolean whether to edit the response or not? If that's the case we would explicitly check the type of the return value to see if it's boolean instead of converting it to a boolean.

Reply:
If you implement both, you'll end up with 2.0.0 instead of 1.1.0 if you implement only once.
Besides, they conflict with each other.

Reply:
Then can't we handle that conflict? A couple conditions on whether the response should be edited:

- Action's boolean is `true` AND index hasn't changed THEN do not edit response.
- Action's boolean is `false` AND index hasn't changed THEN edit response.
- Action's boolean is `true` AND index has changed THEN do not edit response.
- Action's boolean is `false` AND index has changed THEN edit response.
- Action's boolean is invalid AND index hasn't changed THEN do not edit response.
- Action's boolean is invalid AND index has changed THEN edit response.

In other words if the boolean is `true` then do NOT edit the response, otherwise edit.

Reply:
Please don't try to make this API any harder than my Intelligent Behaviour System's practices, keep it simple, just edit when index was changed.

Reply:
Alright, I'll change that.